sct-commons/src/main/java/org/lfenergy/compas/sct/commons/util/FcdaCandidatesHelper.java

```diff
@@ -0,0 +1,105 @@
+// SPDX-FileCopyrightText: 2022 RTE FRANCE
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package org.lfenergy.compas.sct.commons.util;
+
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import org.apache.commons.lang3.StringUtils;
+import org.lfenergy.compas.sct.commons.dto.ResumedDataTemplate;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public enum FcdaCandidatesHelper implements Predicate<ResumedDataTemplate> {
+    SINGLETON;
+
+    private static final String FCDA_CONSTRAINTS_FILE_NAME = "FcdaCandidates.csv";
+    public static final int NUMBER_OF_FIELDS = 4;
+    private static final int FIELD_LN_CLASS = 0;
+    private static final int FIELD_DO_NAME = 1;
+    private static final int FIELD_DA_NAME = 2;
+    private static final int FIELD_FC = 3;
+    private static final String DELIMITER = ";";
+    private static final String COMMENT = "#";
+
+    private Set<FcdaCandidate> fcdaCandidates;
+
+    /**
+     * Alias of isFcdaCandidate(ResumedDataTemplate) to implement Predicate
+     * @see #isFcdaCandidate(ResumedDataTemplate)
+     */
+    @Override
+    public boolean test(ResumedDataTemplate resumedDataTemplate) {
+        return isFcdaCandidate(resumedDataTemplate);
+    }
+
+    /**
+     * Check if resumedDataTemplate is isFcdaCandidate
+     * @see #isFcdaCandidate(ResumedDataTemplate)
+     */
+    public boolean isFcdaCandidate(ResumedDataTemplate resumedDataTemplate) {
```

Comment:
It could be. The FCDA Candidate CSV file is specific to the service that create Dataset and ControlBlock. I prefer FcdaCandidatesHelper to be a dependency of the service instead of ResumedDataTemplate.
What do you think ?

Reply:
I get the point. OK good for me!